curriculum/challenges/english/25-front-end-development/lab-book-inventory-app/66a207974c806a19d6607073.md

```diff
@@ -363,56 +363,82 @@ You should use an attribute selector to target the `span` elements which are dir
+  'span[class~="two"] span:first-child, span[class~="two"] span:nth-child(2)',
+  'span[class~="two"] span:nth-child(2), span[class~="two"] span:first-child',
+  'span[class~="two"] span:nth-of-type(-n+2)',
+  'span[class~="two"] span:nth-child(-n+2)'
+  ]
+assert.isNotNull(new __helpers.CSSHelp(document).getStyleAny(selectors));
 ```
 
-You should use an attribute selector to target the first two descendants of `span` elements that have `two` as a part of their `class` value and set their `background-image` property to use a `linear-gradient`.
+You should use an attribute selector to target the first two descendants of `span` elements that have the word `two` as a part of their `class` value and set their `background-image` property to use a `linear-gradient`.
 
 ```js
-const selector1 = new __helpers.CSSHelp(document).getStyle('span[class~="two"] :nth-child(1), span[class~="two"] :nth-child(2)');
-const selector2 = new __helpers.CSSHelp(document).getStyle('span[class~="two"] :nth-child(2), span[class~="two"] :nth-child(1)');
-const selector3 = new __helpers.CSSHelp(document).getStyle('span[class~="two"] :first-child, span[class~="two"] :nth-child(2)');
-const selector4 = new __helpers.CSSHelp(document).getStyle('span[class~="two"] :nth-child(2), span[class~="two"] :first-child');
-const selectors = [selector1, selector2, selector3, selector4];
-let containsSelector = false;
-for (let selector of selectors) {
-  if (selector?.backgroundImage.includes('linear-gradient(')) {
-    containsSelector = true;
-  }
-}
-assert.isTrue(containsSelector); 
+const selectors = [
+  'span[class~="two"] :nth-child(1), span[class~="two"] :nth-child(2)',
+  'span[class~="two"] :nth-child(2), span[class~="two"] :nth-child(1)',
+  'span[class~="two"] :first-child, span[class~="two"] :nth-child(2)',
+  'span[class~="two"] :nth-child(2), span[class~="two"] :first-child',
+  'span[class~="two"] :nth-of-type(-n+2)',
+  'span[class~="two"] :nth-child(-n+2)',
+  'span[class~="two"] span:nth-child(1), span[class~="two"] span:nth-child(2)',
+  'span[class~="two"] span:nth-child(2), span[class~="two"] span:nth-child(1)',
+  'span[class~="two"] span:first-child, span[class~="two"] span:nth-child(2)',
+  'span[class~="two"] span:nth-child(2), span[class~="two"] span:first-child',
+  'span[class~="two"] span:nth-of-type(-n+2)',
+  'span[class~="two"] span:nth-child(-n+2)'
+  ]
+
+assert.isTrue(new __helpers.CSSHelp(document).getStyleAny(selectors)?.backgroundImage.includes('linear-gradient(')) 
 ```
 
-You should have an attribute selector to target the `span` elements that are descendants of `span` elements that have `three` as a part of their `class` value.
+You should have an attribute selector to target the `span` elements that are descendants of `span` elements that have the word `three` as a part of their `class` value.
 
 ```js
 assert.exists(new __helpers.CSSHelp(document).getStyle('span[class~="three"] span'));
```

Comment:
Do these ones not need to allow for additional selector combinations?

Reply:
There are 3 `span` elements there, and campers should target all of them. So I think this should be the accepted selector. But if you think we should accept other solutions I can add them.